### PR TITLE
Q package removal extension folder

### PR DESCRIPTION
**Pull request by SounD120**
#1113

### CODE REVIEW

File: src/extension/android/adb.ts [hunk 1]
```diff
@@ -75,19 +73,19 @@ export class AdbHelper {
...
-                let deferred = Q.defer();
-                setTimeout(() => {
-                    this.stopApp(projectRoot, packageName, debugTarget)
-                        .then(() => {
-                            return deferred.resolve({});
-                        });
-                }, 200); // We need a little delay after broadcast command
-
-                return deferred.promise;
+                let deferred = new Promise((resolve) => {
```
Comment (RedMickey) [inline src/extension/android/adb.ts:80]: Can we return just a promise without creating the variable here?

File: src/extension/android/androidPlatform.ts [hunk 2]
```diff
@@ -187,21 +185,22 @@ export class AndroidPlatform extends GeneralMobilePlatform {
...
+            return this.configureADBReverseWhenApplicable(device);
+            })
             .then(() => {
-                return this.configureADBReverseWhenApplicable(device);
-            }).then(() => {
                 return this.needsToLaunchApps
                     ? this.adbHelper.launchApp(this.runOptions.projectRoot, this.packageName, device.id)
-                    : Q<void>(void 0);
-            }).then(() => {
+                    : Promise.resolve();
```
Comment (RedMickey) [inline src/extension/android/androidPlatform.ts:195]: Could it be better to use the `resolve ()` function instead of `Promise.resolve()`?

Comment (SounD120) [reply src/extension/android/androidPlatform.ts:193]: We cannot pass `resolve` function from the top Promise to `then`.

File: src/extension/ios/iOSPlatform.ts [hunk 3]
```diff
@@ -114,27 +113,27 @@ export class IOSPlatform extends GeneralMobilePlatform {
                 this.runArguments.push("--verbose");
             }
             const runIosSpawn = new CommandExecutor(this.projectPath, this.logger).spawnReactCommand("run-ios", this.runArguments, {env});
-            return new OutputVerifier(() => this.generateSuccessPatterns(this.runOptions.reactNativeVersions.reactNativeVersion), () => Q(IOSPlatform.RUN_IOS_FAILURE_PATTERNS), "ios")
+            return new OutputVerifier(() => this.generateSuccessPatterns(this.runOptions.reactNativeVersions.reactNativeVersion), () => Promise.resolve(IOSPlatform.RUN_IOS_FAILURE_PATTERNS), "ios")
```
Comment (RedMickey) [inline src/extension/ios/iOSPlatform.ts:116]: Could it be better to split this one long string into several ones?

**Pull request merged**